$NetBSD: patch-bu,v 1.1 2011/08/09 07:15:42 manu Exp $

Avoid fatal error if gfid changed. Patch not committed upstream yet,
but it helps.

--- xlators/protocol/client/src/client3_1-fops.c.orig	2011-07-14 14:21:10.000000000 +0200
+++ xlators/protocol/client/src/client3_1-fops.c	2011-07-22 09:38:35.000000000 +0200
@@ -2208,13 +2208,11 @@
         }
 
         if ((!uuid_is_null (inode->gfid))
             && (uuid_compare (stbuf.ia_gfid, inode->gfid) != 0)) {
-                gf_log (frame->this->name, GF_LOG_DEBUG,
-                        "gfid changed for %s", local->loc.path);
-                rsp.op_ret = -1;
-                op_errno = ESTALE;
-                goto out;
+		gf_log (frame->this->name, GF_LOG_WARNING,
+			"gfid changed for %s", local->loc.path);
+		memcpy(&stbuf.ia_gfid, &inode->gfid, sizeof(inode->gfid));
         }
 
         rsp.op_ret = 0;
 
